pepper-apis/src/test/java/org/broadinstitute/ddp/service/WorkflowServiceTest.java

```diff
@@ -360,6 +372,136 @@ public void testBuildStateResponse_activityState_providesActivityMetadata() {
         });
     }
 
+    @Test
+    public void test_givenNullMaxInstPerUser_andNextActDoesntHaveInstance_whenSuggestNextStateIsCalled_thenInstanceIsCreated() {
+        TransactionWrapper.useTxn(handle -> {
+            Map<String, Long> actParticipatingInTransition = setupActivitiesAndTransitionBetweenThem(handle);
+            service.suggestNextState(
+                    handle, operatorGuid, userGuid, studyGuid, new ActivityState(
+                            actParticipatingInTransition.get("activityTransitionedFromId")
+                    )
+            );
+
+            Optional<String> latestInstanceGuid = handle.attach(JdbiActivityInstance.class)
+                    .findLatestInstanceGuidByUserGuidAndActivityId(userGuid, actParticipatingInTransition.get("activityTransitionedToId"));
+            assertTrue(latestInstanceGuid.isPresent());
+            handle.rollback();
+        });
+    }
```

Comment:
This test and a bunch of similar tests below make sure that a new instance is created/not created, depending on the `maxInstancePerUser` values.